### PR TITLE
feat: Create the `aggregator` module

### LINKED ISSUE #269: Split Aggregator into Native and Halo2 version
Native version:
https://github.com/eigen-trust/protocol/blob/master/circuit/src/verifier/aggregator.rs#L65-L106
https://github.com/eigen-trust/protocol/blob/master/circuit/src/verifier/aggregator.rs#L171-L222

Halo2 version:
https://github.com/eigen-trust/protocol/blob/master/circuit/src/verifier/aggregator.rs#L108-L139C1
https://github.com/eigen-trust/protocol/blob/master/circuit/src/verifier/aggregator.rs#L243-L409

Make folder named `aggregator`:
- `native.rs`
- `mod.rs`

EDIT:
We also want to make a native version of aggregator chip, where the logic of aggregation is done using NativeTranscriptRead and NativeLoader.
Reference logic:
https://github.com/eigen-trust/protocol/blob/a3dbc1002d45ccbb57d404e611e00de1d1abcf4b/circuit/src/verifier/aggregator.rs#L325-L400

### PR DESCRIPTION
**Pull request by duguorong009**
## Description
- Split the code of `aggregator` to halo2 part & native one
- Create the `aggregator` module 

Closes #269 

## Changes
- Create new directory `aggregator` in `circuit` 
- Create `native.rs` and `mod.rs` for native version and halo2 version

### CODE REVIEW

File: circuit/src/aggregator/native.rs [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+		for (i, instances) in snark_instances.iter().enumerate().take(self.snarks.len()) {
+			let _: Vec<Vec<()>> = self.snarks[i]
+				.instances
+				.iter()
+				.zip(instances.iter())
+				.map(|(x, y)| {
+					x.iter().zip(y.iter()).map(|(a, b)| a.assert_if_known(|a| a == b)).collect()
+				})
+				.collect();
+		}
```
Comment (lazovicff) [inline circuit/src/aggregator/native.rs:145]: We dont really need this extra step of passing instances. We can just use `self.snarks[i].instances`

**Comment by lazovicff** [comment]
We should further divide native and halo2 version, by having a separate native struct:
```rust
```

File: circuit/src/aggregator/mod.rs [hunk 2]
```diff
@@ -101,6 +101,18 @@ struct Aggregator {
 	as_proof: Option<Vec<u8>>,
 }
 
+impl Aggregator {
+	/// Create a new aggregator.
+	pub fn new(svk: Svk, snarks: Vec<Snark>, instances: Vec<Fr>, as_proof: Vec<u8>) -> Self {
+		Self {
+			svk,
+			snarks: snarks.into_iter().map_into().collect(),
+			instances,
```
Comment (lazovicff) [inline circuit/src/aggregator/mod.rs:110]: We dont need instances here. They are passed to the verifier as public inputs

Comment (duguorong009) [reply circuit/src/aggregator/mod.rs:110]: Ah, yes.
Will update it now!

**Review by lazovicff — APPROVED** [review]

**Pull request merged**